packages/core/src/validate.ts

```diff
@@ -134,15 +166,51 @@ export function getStorageLayout(validation: Validation, version: Version): Stor
   return layout;
 }
 
-export function assertUpgradeSafe(validation: Validation, version: Version, unsafeAllowCustomTypes = false): void {
+function getUnlinkedBytecode(validation: Validation, bytecode: string): string {
```

Comment:
If a match isn't found, this function should throw an error.

Reply:
This does not apply anymore based on the new design.